extension/agenthealth/handler/stats/handler.go

```diff
@@ -21,12 +21,25 @@ const (
 	headerKeyAgentStats = "X-Amz-Agent-Stats"
 )
 
-func NewHandlers(logger *zap.Logger, cfg agent.StatsConfig) ([]awsmiddleware.RequestHandler, []awsmiddleware.ResponseHandler) {
-	filter := agent.NewOperationsFilter(cfg.Operations...)
+func NewHandlers(logger *zap.Logger, cfg agent.StatsConfig, statuscodeonly bool) ([]awsmiddleware.RequestHandler, []awsmiddleware.ResponseHandler) {
+	statusCodeFilter := agent.NewStatusCodeOperationsFilter()
+	if statuscodeonly {
+		statusCodeStats := provider.GetStatusCodeStats(statusCodeFilter)
```

Comment:
In both branches (statuscodeonly true or false), you are executing this line so you can move it outside the if block